app/__init__.py

```diff
@@ -50,4 +56,3 @@ def server_internal_error(message):
     app.register_error_handler(500, server_internal_error)
```

Comment:
In the decorator, you are having 404 while the return statement has 500, which status code is correct here?